src/coreclr/vm/object.cpp

```diff
@@ -1545,13 +1558,11 @@ void StackTraceArray::CopyFrom(StackTraceArray const & src)
     }
     CONTRACTL_END;
 
-    m_array = (I1ARRAYREF) AllocatePrimitiveArray(ELEMENT_TYPE_I1, static_cast<DWORD>(src.Capacity()));
-
-    Volatile<size_t> size = src.Size();
+    Volatile<uint32_t> size = src.Size();
     memcpyNoGCRefs(GetRaw(), src.GetRaw(), size * sizeof(StackTraceElement) + sizeof(ArrayHeader));
-
     SetSize(size);  // set size to the exact value which was used when we copied the data
                     // another thread might have changed it at the time of copying
+    SetKeepAliveItemsCount(ComputeKeepAliveItemsCount());
```

Comment:
@jkotas I am considering making a change here. If I added a way to get and set the size and keep alive items count atomically together from the stacktrace, I could get rid of the ComputeKeepAliveItemsCount usage for anything else than asserts.

Reply:
I think you will have troubles with alignment on 32-bit platforms to make this work.

Reply:
Hmm, right.